js/lex.go

```diff
@@ -374,46 +377,57 @@ func (l *Lexer) consumeSingleLineComment() {
 
 ////////////////////////////////////////////////////////////////
 
-func (l *Lexer) consumeCommentToken() bool {
+func (l *Lexer) consumeHTMLLikeCommentToken() bool {
+	c := l.r.Peek(0)
+	if c == '<' && l.r.Peek(1) == '!' && l.r.Peek(2) == '-' && l.r.Peek(3) == '-' {
+		// opening HTML-style single line comment
+		l.r.Move(4)
+		l.consumeSingleLineComment()
+	} else if l.emptyLine && c == '-' && l.r.Peek(1) == '-' && l.r.Peek(2) == '>' {
+		// closing HTML-style single line comment
+		// (only if current line didn't contain any meaningful tokens)
+		l.r.Move(3)
+		l.consumeSingleLineComment()
+	} else {
+		return false
+	}
+	return true
+}
+
+func (l *Lexer) consumeRegularCommentToken() TokenType {
 	c := l.r.Peek(0)
 	if c == '/' {
 		c = l.r.Peek(1)
 		if c == '/' {
-			// single line
+			// single line comment
 			l.r.Move(2)
 			l.consumeSingleLineComment()
+			return CommentToken
 		} else if c == '*' {
-			// multi line
+			// block comment (potentially multiline)
+			tt := CommentToken
 			l.r.Move(2)
 			for {
 				c := l.r.Peek(0)
 				if c == '*' && l.r.Peek(1) == '/' {
 					l.r.Move(2)
-					return true
+					break
 				} else if c == 0 {
 					break
 				} else if l.consumeLineTerminator() {
+					tt = MultiLineCommentToken
 					l.emptyLine = true
 				} else {
 					l.r.Move(1)
 				}
 			}
+			return tt
 		} else {
-			return false
+			return UnknownToken
```

Comment:
No need for this `else` and the one below, just return `UnknownToken` at the end.